lib/src/bindings/structs.dart

```diff
@@ -118,3 +118,76 @@ class OBX_bytes_array extends Struct {
     return result;
   }
 }
+
+class OBX_int8_array extends Struct {
+  Pointer<Uint8> _itemsPtr;
+
+  @IntPtr() // size_t
+  int count;
+
+  List<int> items() => _itemsPtr.asTypedList(count).toList();
+}
+
+class OBX_int16_array extends Struct {
+  Pointer<Uint16> _itemsPtr;
+
+  @IntPtr() // size_t
+  int count;
+
+  List<int> items() => _itemsPtr.asTypedList(count).toList();
+}
+
+class OBX_int32_array extends Struct {
+  Pointer<Uint32> _itemsPtr;
+
+  @IntPtr() // size_t
+  int count;
+
+  List<int> items() => _itemsPtr.asTypedList(count).toList();
+}
+
+class OBX_int64_array extends Struct {
+  Pointer<Uint64> _itemsPtr;
+
+  @IntPtr() // size_t
+  int count;
+
+  List<int> items() => _itemsPtr.asTypedList(count).toList();
+}
+
+class OBX_string_array extends Struct {
+
+  Pointer<Pointer<Uint8>> _itemsPtr;
+
+  @IntPtr() // size_t
+  int count;
+
+  List<String> items() {
+    final list = <String>[];
+    for (int i=0; i<count; i++) {
+      list.add(Utf8.fromUtf8(_itemsPtr.elementAt(i).value.cast<Utf8>()));
+    }
+    return list;
+  }
+}
+
+class OBX_float_array extends Struct {
+
+  Pointer<Float> _itemsPtr;
+
+  @IntPtr() // size_t
+  int count;
+
+  List<double> items() => _itemsPtr.asTypedList(count).toList();
+}
+
+
+class OBX_double_array extends Struct {
+
+  Pointer<Double> _itemsPtr;
+
+  @IntPtr() // size_t
+  int count;
+
+  List<double> items() => _itemsPtr.asTypedList(count).toList();
+}
```

Comment:
These could probably be replaced with a single template (generic)

Reply:
Tried this, but Struct can't be extended by a generic class. Using a mixin then. But then `.asTypedList(count)` is not available. Anything else in mind?